lib/vintage_net_mobile/modem/automatic/discovery.ex

```diff
@@ -0,0 +1,120 @@
+defmodule VintageNetMobile.Modem.Automatic.Discovery do
+  @moduledoc false
+
+  # A process for trying to discover a the type of modem and some basic
+  # information to allow dynamic configuration items
+
+  use GenServer
+  require Logger
+
+  alias Circuits.UART
+
+  # max backoff for trying to enumerate the UARTS
+  @max_backoff 3_000
+
+  @quectel_vender_id 0x2C7C
```

Comment:
While I definitely don't think we should add all of VID:PIDs in [option.c](https://elixir.bootlin.com/linux/latest/source/drivers/usb/serial/option.c#L245), I think we can proactive add the Quectel ones that we know work with the `QuectelEC25` module and the other ones for here. Perhaps also a note that `option.c` is a good source of VID:PIDs.